Bottom: 67a632edad5dd3fdbbd06a6f851910e7cda522ef
Top:    2c04474d390244ea4d7a79af138444fee83b9004
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-08-27 14:59:07 +0100

Refresh of hid-gtk-encapsulate-view-param

---

diff --git a/src/hid/gtk/gui.h b/src/hid/gtk/gui.h
index 6fe3a04..37251db 100644
--- a/src/hid/gtk/gui.h
+++ b/src/hid/gtk/gui.h
@@ -145,6 +145,19 @@ GhidGui;
 
 extern GhidGui _ghidgui, *ghidgui;
 
+typedef struct
+{
+  double coord_per_px; /* Zoom level described as PCB units per screen pixel */
+
+  Coord x0;
+  Coord y0;
+  Coord width;
+  Coord height;
+
+  bool flip_x;
+  bool flip_y;
+
+} view_data;
 
   /* The output viewport
    */
@@ -168,15 +181,8 @@ typedef struct
   gboolean has_entered;
   gboolean panning;
 
-/* zoom value is PCB units per screen pixel.  Larger numbers mean zooming
-|  out - the largest value means you are looking at the whole board.
-*/
-  gdouble zoom;			/* PCB units per screen pixel.  Larger */
-                                /* numbers mean zooming out. */
-  /* Viewport in PCB coordinates */
-  Coord view_x0, view_y0, view_width, view_height;
+  view_data view;
   Coord pcb_x, pcb_y;
-
   gint crosshair_x, crosshair_y;
 }
 GHidPort;
